WordPress/Classes/ViewRelated/Themes/ThemeBrowserViewController.swift

```diff
@@ -909,3 +869,71 @@ public protocol ThemePresenter: class {
         presentingTheme = nil
     }
 }
+
+// MARK: - NoResults Handling
+
+private extension ThemeBrowserViewController {
+
+    func updateResults() {
+        if themeCount == 0 && customThemeCount == 0 {
+            showNoResults()
+        } else {
+            hideNoResults()
+        }
+    }
+
+    func setupNoResultsViewController() {
+        let noResultsStoryboard = UIStoryboard(name: "NoResults", bundle: nil)
+        guard let noResultsViewController = noResultsStoryboard.instantiateViewController(withIdentifier: "NoResults") as? NoResultsViewController else {
+            return
+        }
+
+        self.noResultsViewController = noResultsViewController
+    }
+
+    func showNoResults() {
+
+        guard !noResultsShown else {
+            return
+        }
+
+        if noResultsViewController == nil {
+            setupNoResultsViewController()
+        }
+
+        guard let noResultsViewController = noResultsViewController else {
+            return
+        }
+
+        let title = searchController.isActive ? NoResultsTitles.noThemes : NoResultsTitles.fetchingThemes
+        noResultsViewController.configure(title: title)
+
+        addChildViewController(noResultsViewController)
+        collectionView.addSubview(noResultsViewController.view)
+        noResultsViewController.view.frame = collectionView.frame
+
+        // There is a gap between the search bar and the collection view - https://github.com/wordpress-mobile/WordPress-iOS/issues/9730
```

Comment:
Nice catch and follow through.  Just to be extra cautious its not accidentally over looked in the future, could we add a comment to https://github.com/wordpress-mobile/WordPress-iOS/issues/9730 noting that this work around needs to be cleaned up as a part of the patch for that issue?

Reply:
Ah, yes, good suggestion. It shall be done!